opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -2290,6 +2290,73 @@ namespace detail {
 
 
 
+
+    template <class Grid, class WellModel, class Implementation>
+    std::vector<V>
+    BlackoilModelBase<Grid, WellModel, Implementation>::
+    computeFluidInPlace(const ReservoirState& x,
+                        const WellState& xw,
+                        const std::vector<int>& fipnum)
+    {
+        using namespace Opm::AutoDiffGrid;
+        const int nc = numCells(grid_);
+        const int np = x.numPhases();
+
+        SolutionState state(np);
```

Comment:
I had thought about this, actually FIP computation only need re_[phase].accum[0], rs and rv should not be added and stored separately. @atgeirr Do you have any plan to refactor this?